lib/index.js

```diff
@@ -55,82 +55,79 @@ function setupRateLimitEventExt(server, options) {
   const extractKey = options.extractKey;
   const onError = options.onError;
 
-  const extractKeyAndTakeToken = function(limitd, request, reply, type) {
-    extractKey(request, reply, (err, key) =>{
-      if (err) { return reply(err); }
-
-      if (!limitd) {
-        // limitd is not connected, do not fail!
-        return reply.continue();
-      }
-
-      limitd.take(type, key, (err, currentLimitResponse) => {
-        if (err){
-          if (onError) { return onError(err, reply); }
-          // by default we don't fail if limitd is unavailable
-          return reply.continue();
-        }
-
-        const oldMinimumLimitResponse = request.plugins.patova && request.plugins.patova.limit
-        const newMinimumLimitResponse = getMinimumLimit(currentLimitResponse, oldMinimumLimitResponse)
-
-        request.plugins.patova = request.plugins.patova || {};
-        request.plugins.patova.limit = newMinimumLimitResponse;
-
-        if (newMinimumLimitResponse.conformant) {
-          // We continue only if the request is conformat so far
-          return reply.continue();
-        }
-
-        const error = Boom.tooManyRequests();
-        error.output.headers = new RateLimitHeaders(
-          newMinimumLimitResponse.limit,
-          newMinimumLimitResponse.remaining,
-          newMinimumLimitResponse.reset);
-
-        reply(error);
-      });
-    });
+  const extractKeyAndTakeToken = async function(limitd, request, handler, type) {
```

Comment:
Maybe use `h` instead of `handler`? Seems it's the convention on the rest of the code.